gutenberg.php

```diff
@@ -71,3 +71,53 @@ function gutenberg_pre_init() {
 
 	require_once __DIR__ . '/lib/load.php';
 }
+
+/**
+ * Displays a warning that deactivating Gutenberg is not recommended when the theme depends
+ * on the FSE functionality found in the plugin.
+ *
+ * @since 10.9.0
+ *
+ * @param string $plugin_file Path to the plugin file relative to the plugins directory.
+ * @param array  $plugin_data An array of plugin data.
+ */
+function gutenberg_theme_relies_on_notice( $plugin_file, $plugin_data ) {
+	if ( ! current_user_can( 'update_plugins' ) ) {
+		return;
+	}
+
+	$theme      = wp_get_theme();
+	$theme_tags = $theme->get( 'Tags' );
+
+	// Only display a notice for themes that require FSE.
+	if ( empty( $theme_tags ) || ! in_array( 'full-site-editing', $theme_tags, true ) ) {
+		return;
+	}
+
+	$wp_list_table = _get_list_table(
+		'WP_Plugins_List_Table',
+		array(
+			'screen' => get_current_screen(),
+		)
+	);
+
+	printf(
+		'<tr class="plugin-update-tr active" id="%s" data-slug="%s" data-plugin="%s">' .
+		'<td colspan="%s" class="plugin-update colspanchange">' .
+		'<div class="notice inline notice-warning notice-alt"><p>%s</p></div></td></tr>',
+		esc_attr( $plugin_data['slug'] . '-required' ),
+		esc_attr( $plugin_data['slug'] ),
+		esc_attr( $plugin_file ),
+		esc_attr( $wp_list_table->get_column_count() ),
+		__( 'This plugin is required for the active theme to work correctly. Deactivating is not recommended.', 'gutenberg' )
+	);
+	?>
+	<style>
+		.plugins tr[data-slug="gutenberg"] th,
+		.plugins tr[data-slug="gutenberg"] td {
```

Comment:
If it's a must, then probably better to target the specific table-row. 

Specificity is unchanged from the PR to the proposed code below. 

```suggestion
		tr.plugin-update-tr[data-slug="gutenberg"] th,
		tr.plugin-update-tr[data-slug="gutenberg"] td {
```
It looks like adding the class `update` to the plugin's row removes drop shadow automatically without the need for these classes.

![Screen Shot 2021-06-24 at 2 19 06 pm](https://user-images.githubusercontent.com/519727/123202000-25bbb000-d4f7-11eb-9890-c63908bc95ed.png)

However, you still get weird double drop shadows when the warning is present and this notice is required. With our without the custom css above, I can't find a way to hide it. I suspect it's some sort of sibling selector that I can't figure out though.

Edit: second screen shot below :D

Reply:
Second screenshot but correctly cropped this time.

![Screen Shot 2021-06-24 at 2 35 20 pm](https://user-images.githubusercontent.com/519727/123203266-69171e00-d4f9-11eb-8447-a40f9c591a72.png)

One thing that occurred to me is that it may be possible to replace the `deactivate` action with a message directing a user to change themes if they wish to do so.

Reply:
I'm still seeing the line above the upgrade notice when one is available (I'm just hacking the version number to 10.7 to check). It seems to be resolved by a more specific selector (`#gutenberg-required td, #gutenberg-required td`) but that ruins the layout when there is no update available.